src/frequenz/sdk/microgrid/_battery/_status.py

```diff
@@ -197,54 +201,120 @@ def battery_id(self) -> int:
         """
         return self._battery.component_id
 
-    def get_status(self) -> BatteryStatus:
-        """Return status of the battery.
+    async def stop(self) -> None:
+        """Stop tracking battery status."""
+        if self._select is not None:
```

Comment:
you should probably also `cancel_and_await(self._task)` here, before killing the `Select`.

Reply:
Done